src/components/editor/markdown-preview/replace-components/vimeo/vimeo-frame.tsx

```diff
@@ -0,0 +1,48 @@
+import { DomElement } from 'domhandler'
+import React, { ReactElement, useCallback } from 'react'
+import { OneClickEmbedding } from '../one-click-frame/one-click-embedding'
+import { testSingleVideoParagraph, VideoFrameProps } from '../video-util'
+
+const getElementReplacement = (node: DomElement, counterMap: Map<string, number>): (ReactElement | undefined) => {
+  const videoId = testSingleVideoParagraph(node, 'vimeo')
+  if (videoId) {
+    const count = (counterMap.get(videoId) || 0) + 1
+    counterMap.set(videoId, count)
+    return <VimeoFrame key={`vimeo_${videoId}_${count}`} id={videoId}/>
+  }
+}
+
+interface VimeoApiResponse {
+  // eslint-disable-next-line camelcase
```

Comment:
should we add an comment to explain this?

Reply:
comment added